### PR TITLE
Bugfix for multihead classification

### PR DESCRIPTION
**Pull request by JihwanEom**
### This PR contains:
- A bugfix for multihead classification by #220.

### CODE REVIEW

File: torchreid/models/efficient_net_pytcv.py [hunk 1]
```diff
@@ -355,7 +355,7 @@ def __init__(self,
             self.output.add_module("asl", AngleSimpleLinear(
                 in_features=final_block_channels,
                 out_features=self.num_classes))
-        elif 'softmax' == self.loss or 'asl' == self.loss:
+        elif 'softmax' in self.loss or 'asl' in self.loss:
```
Comment (harimkang) [inline torchreid/models/efficient_net_pytcv.py:359]: Based on this code, I'm not sure if self.loss is an iterable object. 
Even the if statements before the elif statements still use '=='. I wonder if this is a bug fix.

**Review by harimkang — CHANGES_REQUESTED** [review]

File: torchreid/models/efficient_net_pytcv.py (hunk 1 above)
Comment (JihwanEom) [reply torchreid/models/efficient_net_pytcv.py:359]: self.loss is incoming to this line as string. ex) 'softmax', 'softmax,asl'

But by the modification from #220 , current code allows and parse only when self.loss is exactly same string. This update does not handle multihead classifciation : it uses self.loss as 'softmax,asl'.

So, we need this modification.

Comment (JihwanEom) [reply torchreid/models/efficient_net_pytcv.py:359]: But I'm also wondering whether the above if line also should be changed from == to 'in'. 
Because this can misunderstanding - if we change above == to 'in', when we run multihead task and define self.loss as 'am_binary', the experiments will runs fine but 'am_binary' will not be actually adopted without any error. 
This is the reason why I didn't change above == to 'in'.

Comment (harimkang) [reply torchreid/models/efficient_net_pytcv.py:359]: That is understood but does not seem appropriate in code.
How about this?
```
elif self.loss in ['softmax', 'asl', 'softmax.asl']:
```

Comment (JihwanEom) [reply torchreid/models/efficient_net_pytcv.py:359]: I think that's fine. Do you think also changing above if line is appropriate?

Comment (harimkang) [reply torchreid/models/efficient_net_pytcv.py:359]: I think it doesn't matter if you change to the same convention with the 'if' statement and 'elif' statement.

Comment (JihwanEom) [reply torchreid/models/efficient_net_pytcv.py:359]: Reflected. Thank you for your suggestion.

All the templates runs fine with all tasks.

Comment (goodsong81) [reply torchreid/models/efficient_net_pytcv.py:359]: Is that comma? Then one more suggestion could be a explicit tokenization by ',' then mathcing exact name with '=='.

**Review by goodsong81 — APPROVED** [review]
LGTM if Harim's point is reflected anyhow.

**Comment by JihwanEom** [comment]
Yes, that's comma. I also found multihead engine splits loss : https://github.com/openvinotoolkit/deep-object-reid/blob/ote/torchreid/engine/image/multihead.py#L48

File: torchreid/models/efficient_net_pytcv.py (hunk 1 above)
Comment (JihwanEom) [reply torchreid/models/efficient_net_pytcv.py:358]: Yes, that's comma. I also found multihead engine splits loss : https://github.com/openvinotoolkit/deep-object-reid/blob/ote/torchreid/engine/image/multihead.py#L48

File: torchreid/models/efficient_net_pytcv.py [hunk 2]
```diff
@@ -341,7 +341,8 @@ def __init__(self,
                         tf_mode=tf_mode))
                 in_channels = out_channels
             self.features.add_module("stage{}".format(i + 1), stage)
-        activation = activation if self.loss == 'softmax' else lambda: nn.PReLU(init=0.25)
+        self.loss = self.loss.split(',') if ',' in self.loss else self.loss
+        activation = activation if 'softmax' in self.loss else lambda: nn.PReLU(init=0.25)
```
Comment (JihwanEom) [inline torchreid/models/efficient_net_pytcv.py:345]: @supersoob @harimkang @goodsong81  I found some suspicious code, self.loss is defined as 'softmax,asl' in multihead task, so I confirmed activation function is always defined as swish(default), not PReLU. Could you review this line?

File: torchreid/models/efficient_net_pytcv.py [hunk 3]
```diff
@@ -341,7 +341,8 @@ def __init__(self,
                         tf_mode=tf_mode))
                 in_channels = out_channels
             self.features.add_module("stage{}".format(i + 1), stage)
-        activation = activation if self.loss == 'softmax' else lambda: nn.PReLU(init=0.25)
```
Comment (JihwanEom) [inline torchreid/models/efficient_net_pytcv.py:344]: @supersoob @harimkang @goodsong81 I found some suspicious code, self.loss is defined as 'softmax,asl' in multihead task, so I confirmed activation function is always defined as swish(default), not PReLU. Could you review this line?

File: torchreid/models/efficient_net_pytcv.py (hunk 1 above)
Comment (JihwanEom) [reply torchreid/models/efficient_net_pytcv.py:359]: Revised with below code:
`self.loss = self.loss.split(',') if ',' in self.loss else self.loss`

Comment (harimkang) [reply torchreid/models/efficient_net_pytcv.py:359]: String returns a list if you use split, so why don't you think of 'self.loss' as a list and match the rest of the inspection code?
`self.loss = self.loss.split(',')` returns the list without the need for an if & else statement.

Comment (JihwanEom) [reply torchreid/models/efficient_net_pytcv.py:359]: Updated. Thank you.

**Review by harimkang — APPROVED** [review]
